### ISSUE TITLE
Implement native ruby parser

### ISSUE THREAD

**Issue by flbulgarelli** [mumuki/mulang#214]
Although Ruby support #59 has been implemented externally, it would be nice to actually have a Haskell parser, which may be faster, simpler to use, and may be bundled within the JS distribution, making it more portable.

The bad news is that there is no `Language.Ruby` package over there. 
The good news is that https://github.com/whitequark/parser seems to be implemented using yacc - https://github.com/whitequark/parser/blob/master/lib/parser/ruby23.y - and that happy - which we are indirectly using with some of our own parsers - is  inspired on it, so perhaps porting the BNF may not be that difficult nor a complete rewrite.

**Comment by flbulgarelli** [comment]
Also, we could get rid off that ugly warning @julian-berbel https://github.com/whitequark/parser/issues/286 :stuck_out_tongue:

**Comment by flbulgarelli** [comment]
This issue is in-progress here: https://github.com/flbulgarelli/language-ruby

**Comment by luchotc** [comment]
Can we close this one and move the specific issues to that repo? @flbulgarelli @julian-berbel

**Comment by flbulgarelli** [comment]
Nope, since even after implementing that parser, we will have to integrate it into mulang.
